### ISSUE TITLE
Creating a fake teacher model that only returns the pre-saved features/logits based on data index

### ISSUE THREAD

**Issue by YangLi309** [open-mmlab/mmrazor#566]
### Checklist

- I have searched related issues but cannot get the expected help.
- I have read related documents and don't know what to do.

### Describe the question you meet

Hi there,

Due to the high computational complexity, I have to save all feature maps and output logits of the teacher model on the whole train set. I am wondering if it is possible to create a generic teacher that only returns the corresponding results based on the data indexes? My initial thought would be editing mmengine code to return the sampled data indexes. Would be appreciated if any easier solutions can be provided. Thank you in advance!

Regards,
Yang
